src/OmniSharp.MSBuild/ProjectFile/ProjectFileInfo.ProjectData.cs

```diff
@@ -227,7 +234,17 @@ public static ProjectData Create(MSB.Execution.ProjectInstance projectInstance)
                     targetFramework, targetFrameworks,
                     outputKind, languageVersion, allowUnsafeCode, documentationFile, preprocessorSymbolNames, suppressedDiagnosticIds,
                     signAssembly, assemblyOriginatorKeyFile,
-                    sourceFiles, projectReferences, references.ToImmutable(), packageReferences, analyzers);
+                    sourceFiles, projectReferences, references.ToImmutable(), packageReferences, analyzers, ruleset);
+            }
+
+            private static RuleSet ResolveRulesetIfAny(MSB.Execution.ProjectInstance projectInstance)
+            {
+                var rulesetIfAny = projectInstance.Properties.SingleOrDefault(x => x.Name == "ResolvedCodeAnalysisRuleSet");
```

Comment:
Do we want this to throw if there happen to be two properties? That will mean that OmniSharp will essentially not load the project at all. Perhaps we should be resilient and pick the first instead? That seems to be what you intended anyway based on the fact that the variable name says "IfAny"